ttnn/cpp/ttnn/operations/matmul/device/matmul_op_multi_core_reuse_mcast_dram_sharded_program_factory.cpp

```diff
@@ -378,8 +464,10 @@ operation::ProgramWithCallbacks create_program_dram_sharded(
 
     if (device->arch() == tt::ARCH::WORMHOLE_B0) {
         get_dram_reader_core_coords_wormhole_b0(device, all_worker_cores, all_worker_cores_ordered);
-    } else {
+    } else if (device->arch() == tt::ARCH::GRAYSKULL) {
         get_dram_reader_core_coords_grayskull(device, all_worker_cores, all_worker_cores_ordered);
+    } else if (device->arch() == tt::ARCH::BLACKHOLE) {
```

Comment:
It would be good to do something like either changing to
```
} else {
  TT_FATAL(device->arch() == tt::ARCH::BLACKHOLE, "Device not supported");
  ...
```
or adding another clause
```
...
} else {
  TT_THROW("Device not supported");
```
for future proofing.

Reply:
yeah sg, I'll add it